src/views/AddItem.jsx

```diff
@@ -1,3 +1,93 @@
-export function AddItem() {
-	return <p>Hello from the <code>/add-item</code> page!</p>
+import { addItem } from '../api/firebase';
+import { useState } from 'react';
+
+export function AddItem({ listId }) {
+	// declare itemName, daysUntilNextPurchase, and radioSelect
+	// into separate state variables
+	const [itemName, setItemName] = useState('');
+
+	const [daysUntilNextPurchase, setDaysUntilNextPurchase] = useState(null);
+
+	const [radioSelect, setRadioSelect] = useState('soon');
+
+	// Store values inside itemData and sent over to database
+	// otherwise log an error if request fails
+	const onFormSubmit = (event) => {
+		event.preventDefault();
+		const itemData = {
+			itemName,
+			daysUntilNextPurchase,
+		};
+		try {
+			addItem(listId, itemData);
+			alert(`${itemData.itemName} was added to your list`);
+		} catch (error) {
+			console.log(error);
+		}
+	};
+
+	// Get user input and set itemName
+	const handleItemName = (event) => {
+		setItemName(event.target.value);
+	};
+
+	// Get radio selection and set daysUntilNextPurchase
+	const handlePurchaseDate = (event) => {
+		let date = null;
+		if (event.target.value === 'soon') {
```

Comment:
This works great of course. For the future, one thing to consider would be to be setting the value on the inputs directly to the number like this:
```html
<input type="radio" name="radio-btn" value="7" />
```
and then just calling 
```javascript
const date = Number(event.target.value);
```

On the other hand, one advantage of having it the way you do here is that it is probably easier to follow (logic is in one place instead of needing to check it in the markup as well).

Food for thought 🤔

Reply:
Oh wow, we didn't even think of doing it this way! So much easier on the eyes. You do have a point with the logic part. I suppose in larger applications and with forms that have like 20 inputs, it could be hard to keep track of values in markup.